server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/UpdateMobileBuildHelper.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Arcadia.Assistant.AppCenterBuilds.Contracts.AppCenter;
+using Arcadia.Assistant.MobileBuild.Contracts.Interfaces;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Arcadia.Assistant.AppCenterBuilds
+{
+    internal class UpdateMobileBuildHelper
+    {
+        private readonly string buildUrl;
+        private readonly string downloadUrlTemplate;
+        private readonly string apiKey;
+
+        public UpdateMobileBuildHelper(string buildUrl, string downloadUrlTemplate, string apiKey)
+        {
+            this.buildUrl = buildUrl;
+            this.downloadUrlTemplate = downloadUrlTemplate;
+            this.apiKey = apiKey;
+        }
+
+        #region public mehods
+
+        public async Task<bool> CheckAndupdateMobileBuild(IHttpClientFactory httpClientFactory, IMobileBuildActor mobileBuildActor, CancellationToken cancellationToken, Action<string> logStore)
+        {
+            var currentMobileBuildVersion = await mobileBuildActor.GetMobileBuildVersionAsync(cancellationToken);
+            var appCenterLatestBuild = await GetLatestBuild(httpClientFactory);
+            var appCenterLastBuildVersion = appCenterLatestBuild.Id.ToString();
+
+            if (currentMobileBuildVersion != appCenterLastBuildVersion)
+            {
+                var downloadModel = await GetBuildDownloadModel(appCenterLatestBuild, httpClientFactory);
+                var data = await GetBuildData(downloadModel, httpClientFactory);
+                await mobileBuildActor.SetMobileBuildData(appCenterLastBuildVersion, data, cancellationToken);
+                logStore($"Mobile build {appCenterLastBuildVersion} updated from {downloadModel.Uri}");
+            }
+            else
+            {
+                logStore("The same version - nothing to do");
```

Comment:
That's why I mentioned logging as a separate task :) We should either keep this line and refactor it later when you deal with it or just add a todo (which i'd prefer)

Reply:
Added 'TO DO' for some catch places.